versioned_docs/version-stable/nym-simple-websocket-tutorial.md

```diff
@@ -0,0 +1,658 @@
+- Saves a `config.toml`  file on our computers users directory within the folder `.nym` (dont worry too much about this for now).
+- Starts using a gateway on the mixnet.
+- Generates and address for our Nym Websocket Client.
+
+Next, within the same terminal, lets run the following command:
+
+```
+./nym-client run --id websocket-client 
+```
+
+You should then see and output looking something like this:
+
+<img src="/img/tutorials/simple-websocket/image8.png"/>
+
+<br/>
+<br/>
+
+Our Nym Websocket Client for our Typescript Script is now up and running! Let's have a look at what is happening in our Web Application in the browser (you may want to refresh the browser one back on the application).
+
+<img src="/img/tutorials/simple-websocket/image9.png"/>
+
+<br/>
+<br/>
+
+We have a response in our Activity Log section at the bottom of our application. We can see that we got a successful response from our websocket, hence we were able to get back the address we saw in the above terminal output. Lets try something here. Terminate the terminal process by holding `CTRL + C`.  After that, lets take another look at our browser application.
+
+<img src="/img/tutorials/simple-websocket/image10.png"/>
+
+<br/>
+<br/>
+
+Excellent, so we also know that our error code for a missing websocket is also working for us. Lets get that client back up and running (by using `./nym-client run --id websocket-client`)so we can get on to connecting it to a Service Provider.
+
+### Adding our Service Provider
+
+Its time to get a full circle of websocket functionality up and running. Go back to https://github.com/nymtech/developer-tutorials, open up the Websocket Client Demo folder and download our Service Provider code.
+
+Like our User Client , the Service Provider is a typescript application, bundles using parcel and has the same npm dependencies. Once downloaded, place the project in an appropriate folder and navigate to the project folder location in a new terminal window. Make sure your in the project root (the same level as your `package.json`, `tsconfig.json`, etc) and run:
+
+```
+npm install 
+```
+
+Then in your package.json, under the “scripts” section. Check that the “start” line looks like this:
+
+```
+"start": "parcel src/index.html --port 1235",
+```
+
+This will ensure that our User Client and Service Provider wont have a port conflict when we run them locally. Go ahead and then run :
+
+```
+npm start
+```
+
+We should then have the following screen appear when we open a new tab on url : localhost:1235.
+
+<img src="/img/tutorials/simple-websocket/image11.png"/>
+
+<br/>
+<br/>
+
+So we can see that the Service Provider is wanting to listen to a websocket connection on port 1978 but currently it cant find one. That's our cue  to open up another terminal window (in the same location where we have our `nym-client`and `nym-client.d` file) and get ourselves a second instance of our Nym Websocket Client running.
```

Comment:
Are we not going to walk through writing the service provider code like we did with the client side code? Why not?